Module 5/script.sql

```diff
@@ -0,0 +1,176 @@
+IF NOT EXISTS(SELECT * FROM sys.databases WHERE name = 'DeliveryDB')
+  BEGIN
+    CREATE DATABASE [DeliveryDB]
+  END
+GO
+
+USE [DeliveryDB]
+
+IF NOT EXISTS ( SELECT  *
+                FROM    sys.schemas
+                WHERE   name = 'Delivery' )
+    EXEC('CREATE SCHEMA [Delivery]');
+GO
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='Truck' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[Truck]
+	(
+	 [Id]                 int NOT NULL ,
+	 [Brand]              nvarchar(max) NOT NULL ,
+	 [Payload]            float NOT NULL ,
+	 [Volume]             float NOT NULL ,
+	 [FuelConsumption]    float NOT NULL ,
+	 [RegistrationNumber] nvarchar(max) NOT NULL ,
+	 [Year]               date NOT NULL , 
+	)
+END
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='TruckDriver' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[TruckDriver]
+	(
+	 [TruckId]  int NOT NULL ,
+	 [DriverId] int NOT NULL ,
+	);
+
+END
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='Driver' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[Driver]
+	(
+	 [Id]        int NOT NULL ,
+	 [FirstName] nvarchar(max) NOT NULL ,
+	 [LastName]  nvarchar(max) NOT NULL ,
+	 [Birthdate] date NOT NULL ,
+	);
+END
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='State' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[State]
+	(
+	 [Id]   int NOT NULL ,
+	 [Name] nvarchar(max) NOT NULL ,
+	);
+END
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='City' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[City]
+	(
+	 [Id]      int NOT NULL ,
+	 [StateId] int NOT NULL ,
+	 [Name] nvarchar(max) NOT NULL ,
+	);
+END
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='Warehouse' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[Warehouse]
+	(
+	 [Id]     int NOT NULL ,
+	 [CityId] int NOT NULL ,
+	);
+END
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='Route' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[Route]
+	(
+	 [Id]          int            NOT NULL ,
+	 [OriginWarehouseId]      int NOT NULL ,
+	 [DestinationWarehouseId] int NOT NULL ,
+	 [Distance]               float NOT NULL ,
+	);
+END
+
+IF NOT EXISTS (SELECT * FROM sysobjects WHERE name='Shipment' and xtype='U')
+BEGIN
+	CREATE TABLE [Delivery].[Shipment]
+	(
+	 [Id]      int NOT NULL ,
+	 [RouteId] int  NOT NULL ,
+	 [TruckId] int NOT NULL ,
+	 [CargoId] int NOT NULL ,
+	 [Price]   float NOT NULL ,
```

Comment:
"Company’s shipment prices depend on the distance between origin and destination warehouses, the weight (in kilograms) and the volume (m3) of the cargo" - due to this i decided to create such property

Reply:
Yes, look at the following quote: "Company’s shipment prices depend on the distance between origin and destination warehouses, the weight (in kilograms) and the volume (m3) of the cargo." There are nothing about creation such property but probably it is implicit

Reply:
I thought that price would be calculated by some function/sp depending on the cargo. Otherwise, you'll need to have triggers on the Cargo table for update/delete and recalculate prices for all the shipments associated if the volume or weight changes each time.
I'm saying that implementing it as a field seems less practical, although as it hasn't been specified explicitly, it can be left.